service/resolver.go

```diff
@@ -119,40 +119,59 @@ func (r *resolve) MapK8sResourceAthenzResource(k8sRes string) string {
 	return athenzRes
 }
 
-// createAthenzDomain use cfg.ServiceAthenzDomain;
+// createAthenzDomain use cfg.ServiceAthenzDomains;
 
+// do the following for each cfg.ServiceAthenzDomains
 // split it with ".";
 // for each token, if it match /^_.*_$/ but not "_namespace_", replace the token with config.GetActualValue(token);
 // and then return the processed value
-func (r *resolve) createAthenzDomain() string {
-	reps := make([]string, 0, strings.Count(r.cfg.ServiceAthenzDomain, ".")+1)
-	for _, v := range strings.Split(r.cfg.ServiceAthenzDomain, ".") {
-		if v != "_namespace_" && strings.HasPrefix(v, "_") && strings.HasSuffix(v, "_") {
-			// Note: If deploying in a different namespace than the kube-public namespace, change it to get information from kube api
-			reps = append(reps, v, config.GetActualValue(v))
+func (r *resolve) createAthenzDomain() []string {
+	domains := make([]string, 0, len(r.cfg.ServiceAthenzDomains))
+	for _, domain := range r.cfg.ServiceAthenzDomains {
+		reps := make([]string, 0, strings.Count(domain, ".")+1)
```

Comment:
please pre-allocate before domain loop & flush reps here

Reply:
I tried it. e81ef99
Is recognition correct?

Reply:
I think what kpango means is
```
...
for _, domain := range r.cfg.ServiceAthenzDomains {
    reps := make([]string, len(r.cfg.ServiceAthenzDomains))
    ...
```

Reply:
I mean 
```
	reps := make([]string, 0, strings.Count(domain, ".")+1)
	for _, domain := range r.cfg.ServiceAthenzDomains {
		reps = reps[:0]
```

Reply:
@kpango 
I don't think domain can be used because it is not in the domain loop.
For this reason, e81ef99 uses 64 as a cap.
Is there any problem with this?

Reply:
you can count average value using ```(strings.Count(strings.Join(r.cfg.ServiceAthenzDomains, "-"), ".")/len(r.cfg.ServiceAthenzDomains))+1```

Reply:
thank you.

I will use average value.

Reply:
@kpango 
I implemented using this. 29eea3a